src/App.tsx

```diff
@@ -1,27 +1,48 @@
-import React from 'react';
-import './App.scss';
+import React, { useState } from 'react';
+
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
+
+import './App.scss';
+
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[] | null>(null);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={async () => setGoods(await getAll())}
```

Comment:
it is bad practise use anonymous arrow functions in jsx
consider to make one handler which will be accept function and return async arrow function
```
const makeSetGoods = (callback) => async () => setGoods(await callback());
```

```suggestion
        onClick={makeSetGoods(getAll)}
```